src/linux/init/WSLAInit.cpp

```diff
@@ -85,10 +85,18 @@ void HandleMessageImpl(wsl::shared::SocketChannel& Channel, const WSLA_ACCEPT& M
 
     Channel.SendResultMessage<uint32_t>(SocketAddress.svm_port);
 
-    wil::unique_fd Socket{UtilAcceptVsock(ListenSocket.get(), SocketAddress, SESSION_LEADER_ACCEPT_TIMEOUT_MS)};
+    wil::unique_fd Socket{UtilAcceptVsock(ListenSocket.get(), SocketAddress, SESSION_LEADER_ACCEPT_TIMEOUT_MS, Message.Fd != -1)};
     THROW_LAST_ERROR_IF(!Socket);
 
-    THROW_LAST_ERROR_IF(dup2(Socket.get(), Message.Fd) < 0);
+    if (Message.Fd != -1)
```

Comment:
Is there any validation around the file descriptor numbers? What would happen if 0, 1, or 2 was specified?

Reply:
That would be OK. That's actually the mechanism we use when we create a user process (we connect sockets for all std handles)